navigators/UnAuthNavigator.tsx

```diff
@@ -0,0 +1,78 @@
+import {Button, useColorScheme} from 'react-native';
+import {
+  NavigationContainer,
+  type NavigationProp,
+  useNavigation,
+} from '@react-navigation/native';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {useCallback} from 'react';
+
+import {getAppColorStyles} from '../utils/styles/colors';
+import {getAppColors} from '../utils/colors';
+import HomeScreen from '../screens/auth/home/HomeScreen';
+import DetailedMuseumScreen from '../screens/auth/DetailedMuseumScreen';
+import type Museum from '../model/Museum';
+import LoginScreen from '../screens/unauth/LoginScreen';
+import {RegisterScreen} from '../screens/unauth/RegisterScreen';
+
+import {AuthStackNavigation} from './AuthNavigator';
+
+export type UnAuthRootStackParamList = {
+  Login: undefined;
+  Register: undefined;
+};
+
+export type UnAuthStackNavigation = NavigationProp<UnAuthRootStackParamList>;
+
+const Stack = createNativeStackNavigator<UnAuthRootStackParamList>();
+export default function UnAuthNavigator() {
+  const colorScheme = useColorScheme();
+  const {backgroundStyle} = getAppColorStyles(colorScheme);
+  const {backgroundColor, headerColor, textColor} = getAppColors(colorScheme);
+  const navigation = useNavigation<UnAuthStackNavigation>();
+  const navigateToLogin = useCallback(() => {
+    navigation.reset({index: 0, routes: [{name: 'Login'}]});
```

Comment:
Using `navigation.reset` should be the last resort - in this case a simple `navigation.navigate` will be enough :)
```suggestion
    navigation.navigate("Login);
```

Reply:
Do you suggest to do:
navigation.navigate("Login") and navigation.goBack() ?

Reply:
`.navigate` will handle both cases - if the screen has been already on the stack, it will execute `goBack` logic

Reply:
oh, if so perfect!